### ISSUE TITLE
Integer cast before saving the boundary aware offset loss weights

### ISSUE THREAD

**Issue by Tripton** [jsaric/panoptic-swiftnet#3]
Hello,
thanks for the contribution. I am interested in the boundary aware offset loss.
However, I am having some problems understanding a particular integer cast in the script.

As far as I understand, I need to run [this](https://github.com/jsaric/panoptic-swiftnet/blob/main/data/save_offset_weights.py) script to precalculate the weights.
This will create the weight for each image, saves it, and then use it during training.

More precisely, in L78 a transform is used which determines the weights. The returned values "offset_weights" lie in the range of [0, 8] (set by the alphas + void).
In the following line (L79) the offset weights are scaled to the range [0, 1].
```offset_weights = offset_weights / max(transform.alphas)```
Up to this point, I think I understand the script.

However, there seems to be a problem in the next line (L80).
```offset_weights = offset_weights.squeeze().astype(np.uint8)```

As far as I understand, this limits the values to 0 or 1.
Most values within instances are set to 0 and only the border relative to the instance's size is set 1, I am not sure if the loss is as described in the paper. 
Is this a bug in the released code?

**Comment by danielS91** [comment]
Any news on this?

**Comment by jsaric** [comment]
Hello, I apologize for the late response. I was on vacation and was not checking on Github. 

Yes, you are right. I uploaded the wrong version of the script. Line 79 should've been deleted.
What happened is that when debugging I would add this line: 
offset_weights = offset_weights / max(transform.alphas) * 255
It scales the weights to be visible when looking at the grayscale image.
Somehow, prior to uploading, I did not delete the whole line... I am sorry. 

I will fix it in the repo soon, and close the issue.
